girder/api/v1/notification.py

```diff
@@ -105,3 +106,23 @@ def streamGen():
 
                 time.sleep(wait)
         return streamGen
+
+    @access.cookie
+    @access.token
+    @autoDescribeRoute(
+        Description('List notification events')
+        .notes('This endpoint can be used for manual long-polling when '
+               'SSE support is disabled or otherwise unavailable.')
+        .param('since', 'Filter out events before this time stamp.',
+               dataType='integer', required=False)
+        .errorResponse()
+        .errorResponse('You are not logged in.', 403)
+    )
+    def listNotifications(self, params):
+        user, token = self.getCurrentUser(returnToken=True)
+        since = params.get('since')
+        setResponseHeader('Date', str(time.time()))
```

Comment:
As currently written, there is a chance that notifications would be duplicated during polling if they get written between the execution of line 124 and 128. In the stream version, the last update time is set as the maximum of the timestamps on the fetched batch of events. Let's adopt the same approach here, or at least some solution that relies on the database as the authoritative source of truth.

Reply:
(As an aside, those sorts of race conditions will exist regardless because do not generate and write timestamps to the database atomically. If we wanted to fix that, we can use mongo's `$currentDate` operator on writes.)

Reply:
I fixed it the best I could.  We can't always use the same method as the notification stream because there won't always be a notification to send back.  As you said though, these race conditions are possible regardless.